app_feup/lib/view/Widgets/canteen_row.dart

```diff
@@ -0,0 +1,109 @@
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
+            crossAxisAlignment: CrossAxisAlignment.center,
+            children: [
+              Expanded(child: Column(children: getMenuRows(context),))
+            ],
+          ),
+        ));
+  }
+
+  List<Widget> getMenuRows(BuildContext context) => [
+        Container(
+            padding: EdgeInsets.all(12.0),
+            decoration: BoxDecoration(
+                border: Border(
+                    bottom: BorderSide(
+                        width: 0.7, color: Theme.of(context).accentColor))),
+            child: Row(
+                mainAxisAlignment: MainAxisAlignment.center,
+                children: <Widget> [
+                  Icon(Icons.restaurant, size: this.iconSize),
+                  Expanded(
+                      child: Text(this.meatMenu,
+                        textAlign: TextAlign.center
+                      )
+                  ) ,
+              ]
+            )),
+        Container(
+            padding: EdgeInsets.all(12.0),
+            decoration: BoxDecoration(
+                border: Border(
+                    bottom: BorderSide(
+                        width: 0.7, color: Theme.of(context).accentColor))),
+            child: Row(
+                mainAxisAlignment: MainAxisAlignment.center,
+                children: <Widget> [
+                  Icon(Icons.restaurant, size: this.iconSize),
+                  Expanded(
+                      child: Text(this.fishMenu,
+                          textAlign: TextAlign.center
+                      )
+                  ) ,
+              ]
+            )),
+        Container(
+            padding: EdgeInsets.all(12.0),
+            decoration: BoxDecoration(
+                border: Border(
+                    bottom: BorderSide(
+                        width: 0.7, color: Theme.of(context).accentColor))),
+            child: Row(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: <Widget> [
+                  Icon(Icons.restaurant, size: this.iconSize),
+                  Expanded(
+                    child: Text(this.vegetarianMenu,
+                        textAlign: TextAlign.center
+                    )
+                ) ,
+              ]
+            )),
+        Container(
+            padding: EdgeInsets.all(12.0),
+            decoration: BoxDecoration(
+                border: Border(
+                    bottom: BorderSide(
+                        width: 0.7, color: Theme.of(context).accentColor))),
+            child: Row(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: <Widget> [
+                  Icon(Icons.restaurant, size: this.iconSize),
+                  Expanded(
+                       child: Text(this.dietMenu,
+                           textAlign: TextAlign.center
+                    )
+                ) ,
+              ]
+            )),
+      ];
+}
```

Comment:
To make this more maintainable, you could iterate over a list of meals using `forEach`. I think it would be fine to hold a temporary variable to accommodate for more legible code; however, if preparing for different icons for the different meals, changing the class data structure (perhaps using maps for the icons and the meal description) could be a welcome change. What do you think?

Reply:
Done it! Thanks for the tip!